app/features/modelling/galaxy-builder/index.js

```diff
@@ -0,0 +1,197 @@
+        .then(response => response.json())
+        .catch((e) => {
+          console.warn(e);
+        })
+        .then(data => this.handleDataLoad(data));
+    }
+  }
+  handleDataLoad(data) {
+    this.eventHandlers.resizeCanvas({ width: data.width, height: data.height });
+    if (data.psf && data.psfWidth && data.psfHeight) {
+      this.psf = data.psf;
+      this.psfSize = [data.psfWidth, data.psfHeight];
+      this.convolvePSF = convolvePSF(this.regl, this.psf, this.psfSize);
+    }
+    if (data.imageData) {
+      this.imageData = this.regl.texture({
+        type: 'float',
+        channels: 1,
+        data: data.imageData
+      });
+      this.state.shouldCompareToImage = true;
+      this.calculateDifference = calculateDifference(this.regl);
+      this.state.differenceData = new Uint8Array(data.height * data.height * 4);
+    }
+    if (data.imageHeight && data.imageWidth) {
+      this.eventHandlers.changeCanvasStyleSize(
+        { width: `${data.imageWidth}px`, height: `${data.imageHeight}px` }
+      );
+      // this.canvas.style.height = `${data.imageHeight}px`;
+      // this.canvas.style.width = `${data.imageWidth}px`;
+      this.state.sizing = Object.assign(
+        {},
+        this.state.sizing,
+        { height: data.imageHeight, width: data.imageWidth }
+      );
+    }
+    if (data.mask) {
+      this.mask = this.regl.texture({
+        type: 'float',
+        channels: 1,
+        data: data.mask
+      });
+      this.maskImage = maskImage(this.regl);
+    }
+    this.update(this.state.annotations, this.state.sizing);
+    this.eventHandlers.onLoad();
+  }
+  setModel() {
+    // return taskName: render method object
+    this.model = {
+      disk: {
+        name: 'disk',
+        func: drawSersic(this.regl),
+        default: { mux: 0, muy: 0, rx: 10, ry: 15, scale: 5 / 8, roll: 0, i0: 0.75, n: 1, c: 2 }
+      },
+      bulge: {
+        name: 'bulge',
+        func: drawSersic(this.regl),
+        default: { mux: 100, muy: 100, rx: 10, ry: 15, scale: 5 / 8, roll: 0, i0: 0.75, n: 1, c: 2 }
+      },
+      bar: {
+        name: 'bar',
+        func: drawSersic(this.regl),
+        default: { mux: 100, muy: 100, rx: 5, ry: 5, scale: 5 / 8, roll: 0, i0: 0.75, n: 2, c: 2 }
+      },
+      spiral: {
+        name: 'spiral',
+        func: drawSpiral(this.regl),
+        default: { mux: 100, muy: 100, rx: 5, ry: 5, spread: 1, roll: 0, i0: 0.75, n: 2, c: 2, falloff: 1 }
+      }
+    };
+  }
+  calculateModel(annotations, viewBox) {
+    // TODO: store calculated functions in state to be re-called rather than
+    //       re-calculated
+    this.state.annotations = annotations;
+    const s = {
+      size: [this.state.sizing.width, this.state.sizing.height],
+      // the canvas is not the same size as is visible
+      sizeMultiplier: this.canvas.width / this.state.sizing.width,
+      model: this.model
+    };
+    const ret = [];
+    for (let i = 0, comp = null; i < annotations.length; i++) {
```

Comment:
Maybe `Array.map()` here? you could set the variable `ret` to equal `annotations.map(callback)`